### DISCUSSION TITLE
USB-MSC at ESP32-S2

### DISCUSSION THREAD

**Discussion by CmManuES** [hathach/tinyusb#1007] [Q&A]
Hi,
I am running the program example "cdc_msc_freertos" on an ESP32-S2, this is not supplied with voltage via USB, but has its own voltage supply.
It woks fine, the RAM-Disk reports under Windows 10 as drive "TinyUSB MSC (H :)" in Explorer.
However, if I disconnect the USB connection and then reconnect it, the RAM disk does not appear reliably in Explorer. Sometimes it works and sometimes it doesn't.
If the connection is disconnected, the LED flashes in the cycle "BLINK_SUSPENDED". If the USB cable is connected, the LED lights up in the cycle "BLINK_MOUNTED".
I followed the traffic with a USB software sniffer. In the event that the RAM disk does not appear in Explorer, the USB device does not respond to the SCSI command IQUIRY.
The host sends the IQUIRY command:
55 53 42 43 50 3A A9 3B 24 00 00 00 80 00 06 12 00 00 00 24 00 00 00 00 00 00 00 00 00 00 00
and expects 36 bytes as a response, but there is no response from the USB device.
The host waits 20s then tries again.
This also happens if I use the "Eject" command on the RAM disk in Explorer before disconnecting the USB connection. 
Do you have any advice where the problem is?
Thank you for your support.
Kind regards CmManuES

![SCSI-Inquiry](https://user-images.githubusercontent.com/87816226/128631778-f7ad8cb6-a0ef-43c7-83cc-8bf718e3f343.JPG)

**Comment by hathach** [comment]
yeah, esp32s2 disconnection/suspend detection isn't fully implemented yet. I have a todo for this, but haven't got time to work on this.
